docs/source/internals.rst

```diff
@@ -10,3 +10,91 @@ Each notebook extension typically has it's own directory containing:
 * ``thisextension/main.css``: optional CSS
 * ``thisextension/readme.md``: readme file describing the extension in markdown format
 * ``thisextension/config.yaml``: file describing the extension to the ``jupyter_nbextensions_configurator`` server extension
+
+The file names do not need to have the shown names, they can be choosen freely. This is an example for the `main.js` file:
+
+.. code-block:: javascript
+
+    define([
+        'jquery',
+        'base/js/namespace',
+        'base/js/utils',
+        'services/config'
+    ], function (
+        $,
+        IPython,
+        utils,
+        configmod,
+    ) {
+        "use strict";
+
+        // define default values for config parameters
+        var params = {
```

Comment:
rather than having parameters sit in the root of the config file, it can be much neater to group them inside an object named after the extension - see `collapsible_headings` for an example. This also makes the update simpler, as it can be just a `$.extend(true, params, config.data.collapsible_headings);`, since jquery extend is robust to `undefined` values